mirgecom/initializers.py

```diff
@@ -262,6 +263,95 @@ def __call__(self, x_vec, *, t=0, eos=IdealSingleGas()):
                               momentum=mom)
 
 
+class DoubleMachReflection:
+    r"""Implement the double shock reflection problem.
+
+        - Woodward and Collela
+
+    The inital condition is defined
+
+    .. math::
+
+        (\rho,u,v,P) =
+
+    This function only serves as an initial condition
+
+    .. automethod:: __init__
+    .. automethod:: __call__
+    """
+
+    def __init__(
+            self, dim=2, x0=1.0/6.0, us=4.0
+    ):
+        """Initialize initial condition options.
+
+        Parameters
+        ----------
+        dim: int
+           dimension of domain, must be 2
+        x0: float
+           location of shock
+        us: float
+           shock speed
+        """
+        self._x0 = x0
+        self._dim = dim
+        self._us = us
+
+    def __call__(self, x_vec, *, t=0, eos=IdealSingleGas()):
+        """
+        Create the 1D Sod's shock solution at locations *x_vec*.
+
+        Parameters
+        ----------
+        t: float
+            Current time at which the solution is desired (unused)
+        x_vec: numpy.ndarray
+            Nodal coordinates
+        eos: :class:`mirgecom.eos.GasEOS`
+            Equation of state class to be used in construction of soln (if needed)
+        """
+        assert self._dim == 2, "only defined for dim=2"
+
+        gm1 = eos.gamma() - 1.0
+        gmn1 = 1.0 / gm1
+        x_rel = x_vec[0]
+        y_rel = x_vec[1]
+        actx = x_rel.array_context
+
+        zeros = 0*x_rel
+
+        x0 = zeros + self._x0
+        us = zeros + self._us
+        t = zeros + t
+
+        # Mach 4.0
+        rhol = zeros + 4.57142857*1.4
+        rhor = zeros + 1.4
+
+        ul = zeros + 3.125*np.cos(np.pi/6.0)
+        ur = zeros + 0.0
+        vl = zeros - 3.125*np.sin(np.pi/6.0)
+        vr = zeros + 0.0
+        rhoel = zeros + gmn1 * 18.5
+        rhoer = zeros + gmn1 * 1.0
+
+        xinter = (x0 + y_rel/np.sqrt(3.0) + 2.0*us*t/np.sqrt(3.0))
+        sigma = 0.05
+        xtanh = 1.0/sigma*(x_rel-xinter)
+        mass = rhol/2.0*(actx.np.tanh(-xtanh)+1.0)+rhor/2.0*(actx.np.tanh(xtanh)+1.0)
+        rhoe = (rhoel/2.0*(actx.np.tanh(-xtanh)+1.0)
+                + rhoer/2.0*(actx.np.tanh(xtanh)+1.0))
+        u = ul/2.0*(actx.np.tanh(-xtanh)+1.0)+ur/2.0*(actx.np.tanh(xtanh)+1.0)
+        v = vl/2.0*(actx.np.tanh(-xtanh)+1.0)+vr/2.0*(actx.np.tanh(xtanh)+1.0)
+        rhou = mass*u
+        rhov = mass*v
+        energy = rhoe + 0.5*mass*(u*u + v*v)
+
+        return join_conserved(dim=self._dim, mass=mass, energy=energy,
+                              momentum=make_obj_array([rhou, rhov]))
```

Comment:
```suggestion
        vel = make_obj_array([u, v])
        mom = mass * vel
        energy = rhoe + .5*mass*np.dot(vel, vel)

        return join_conserved(dim=self._dim, mass=mass, energy=energy, momentum=mom)
```

What about something like this?

Reply:
Yes, that is better.